lib/SILGen/ManagedValue.h

```diff
@@ -219,20 +219,29 @@ class ManagedValue {
 
   /// Returns true if this managed value can be consumed.
   ///
-  /// This is true if either this value has a cleanup or if it is an
-  /// SSA value without ownership.
+  /// This is true if either this value has a cleanup or if it is a trivial
+  /// object value. For address values, this returns true only if the value has
+  /// a cleanup regardless of whether the type is trivial.
   ///
-  /// When an SSA value does not have ownership, it can be used by a consuming
-  /// operation without destroying it. Consuming a value by address, however,
-  /// deinitializes the memory regardless of whether the value has ownership.
+  /// When an object value is trivial, it can be passed to a consuming operation
+  /// without destroying it. Consuming a value by address, however, always
+  /// deinitializes the memory regardless of its type.
```

Comment:
the memory regardless of its type -> the memory regardless of whether or not it is trivial.